core/src/Pos/Merkle.hs

```diff
@@ -74,48 +76,64 @@ instance (Bi a, Bi (Hash Raw)) => Bi (MerkleTree a) where
     encode = encode . Foldable.toList
     decode = mkMerkleTree <$> decode
 
+newtype MerkleNodeLeft a = MerkleNodeLeft (MerkleNode a)
```

Comment:
What are these `newtype`s gaining? Isn't the position of the argument enough to distinguish them?

Reply:
^ I have the same question

Reply:
Let me be the third to question that :). I really don't think a `newtype` is needed here.